### PR TITLE
feat: log events to PostgreSQL db

### PR DESCRIPTION
**Pull request by rwd**
* Adds a PostgreSQL service dependency, where an events schema holds tables for logging user interaction events with items
  * Added to Docker Compose config, plus pgadmin
  * SQL to init schema is in file `docker/compose/postgres/docker-entrypoint-initdb.d/db.psql`
* Adds middleware to write event logs to new db, and return top trending items based on user activity in the past 24 hours
  * Behind feature toggle `ENABLE_EVENT_LOGGING=1`
* Adds a mixin `logEvent` to post an event to the middleware, used when liking items, viewing items, adding items to a gallery, or downloading an item's media
  * Behind feature toggle `ENABLE_EVENT_LOGGING=1`
* Adds a page at `/trending` to show the trending items

### CODE REVIEW

File: packages/portal/src/server-middleware/api/events/log.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    await client.query(`
+      INSERT INTO events.actions (object_id, action_type_id, session_id, occurred_at)
+      SELECT $1, at.id, $2, current_timestamp
+      FROM events.action_types at
+      WHERE at.name=$3
+      `,
+    [objectRow.id, sessionRow.id, actionType]
+    );
+  } catch (err) {
+    console.error(err);
```
Comment (lbiedinger) [inline packages/portal/src/server-middleware/api/events/log.js:90]: We'll need to log this too.

File: packages/portal/src/mixins/canonicalUrl.js [hunk 2]
```diff
@@ -1,10 +1,11 @@
 export default {
   methods: {
-    canonicalUrl({ locale = true, fullPath = true } = {}) {
-      let path = fullPath ? this.$route.fullPath : this.$route.path;
+    canonicalUrl({ fullPath = true, locale = true, route = null } = {}) {
```
Comment (lbiedinger) [inline packages/portal/src/mixins/canonicalUrl.js:3]: where is the route being passed into canonicalUrl?

File: packages/portal/nuxt.config.js [hunk 3]
```diff
@@ -39,6 +39,22 @@ const redisConfig = () => {
   return redisOptions;
 };
 
+const postgresConfig = () => {
+  const postgresOptions = {
+    enabled: featureIsEnabled('eventLogging'),
+    connectionString: process.env.POSTGRES_URL
```
Comment (lbiedinger) [inline packages/portal/nuxt.config.js:45]: the env variable should probably be POSTGRES_CONNECTION_STRING
We could still have POSTGRES_URL/PORT/USER/PASSWORD set independently if we needed finer level control.

File: packages/portal/src/server-middleware/api/events/log.js (hunk 1 above)
Comment (rwd) [reply packages/portal/src/server-middleware/api/events/log.js:90]: Log to where? The console? `console.error` does log to the console.

File: packages/portal/nuxt.config.js (hunk 3 above)
Comment (rwd) [reply packages/portal/nuxt.config.js:45]: It's been named `POSTGRES_URL` for consistency with our other env vars, and it is a URL, just that `pg` pkg has it named `connectionString`. There is no separate option to `pg` named `url`.

File: packages/portal/src/server-middleware/api/events/log.js (hunk 1 above)
Comment (lbiedinger) [reply packages/portal/src/server-middleware/api/events/log.js]: To the APM I was thinking. Although I see we're not doing that for redis connection errors either, just those would at least respond with an error status in the response.

File: packages/portal/nuxt.config.js (hunk 3 above)
Comment (lbiedinger) [reply packages/portal/nuxt.config.js:45]: a connectionString by any other name would establish a connection as successfully.

**Review by lbiedinger — APPROVED** [review]

**Pull request merged**